scala/scala.bzl

```diff
@@ -494,6 +511,57 @@ def _scala_test_impl(ctx):
     _write_test_launcher(ctx, rjars)
     return _scala_binary_common(ctx, cjars, rjars)
 
+def _prep_grep_pattern_from(patterns):
+    combined_pattern = ""
+    for pattern in patterns:
+      combined_pattern += " -e {}\.class".format(pattern)
+    return combined_pattern
+
+def _discover_classes(ctx, patterns, archive):
+    discovered_classes = ctx.new_file(ctx.label.name + "_discovered_classes.txt")
+    ctx.action(
+      #We need _jdk to even run _jar. Depending on _jar is not enough with sandbox
+      inputs= [archive] + ctx.files._jar + ctx.files._jdk,
```

Comment:
I guess this is supposed to be `ctx.executable._jar` as shown here: https://github.com/bazelbuild/rules_scala/pull/171/files#diff-9957b181001dad28e385437a86e99126R106

then I think we don't need `files._jdk`.